src/roomsyncroniser.ts

```diff
@@ -51,6 +51,28 @@ export class RoomSyncroniser {
 	) {
 		this.roomStore = this.bridge.roomStore;
 		this.mxidLock = new Lock(MXID_LOOKUP_LOCK_TIMEOUT);
+		if (this.bridge.config.metrics.enabled) {
+			const roomMetricsInit = (rooms) => {
+				this.bridge.metrics.room.set(
+					{
+						type: "dm",
+						protocol: this.bridge.protocol.id,
+					},
+					rooms.filter((room) => room.isDirect).length,
+				);
+				this.bridge.metrics.room.set(
+					{
+						type: "group",
+						protocol: this.bridge.protocol.id,
+					},
+					rooms.filter((room) => !room.isDirect).length,
+				);
+			};
+			this.roomStore.getAll()
+				.catch((err) => log.error("could not get room store"))
+				.then(roomMetricsInit)
```

Comment:
You have a race condition here. In the `puppetbridge.ts` initalizing you set `this.bridge.metrics.room` only **after** you created the roomSync object. So you should move the stuff in puppetbridge.ts up a bit, probably

Reply:
Moved that part where `this.bridge.metrics` gets populated into the constructor of the PuppetBridge - see https://github.com/Sorunome/mx-puppet-bridge/pull/72/files#diff-4cd484140d4c36875ceef0af75412976a1090961b49b5137435995661f73cb44R164